packages/api/src/provider-info.ts

```diff
@@ -98,6 +98,12 @@ export interface ProviderInfo {
   // optional creation button title (if defined)
   kubernetesProviderConnectionCreationButtonTitle?: string;
 
+  // indicates if kubernetes provider connection creation is currently disabled
+  kubernetesProviderConnectionCreationDisabled?: boolean;
+
+  // reason why kubernetes provider connection creation is disabled
+  kubernetesProviderConnectionCreationDisabledReason?: string;
```

Comment:
nitpick: this name is consistent with the existing `kubernetesProviderConnectionCreationButtonTitle`but it is a bit long